uberfire/uberfire
# BZ1299619: Business Central Repo Clone Fails To Clone SCP Style SSH URLs

See https://bugzilla.redhat.com/show_bug.cgi?id=1299619

This PR upgrades ```uri.min.js``` to support ```git@github.com:<user>/<repository>.git``` style URL validation and support for SSH Passphrases when cloning. I cannot see any feasible way to write a test for this - although I've confirmed operation with ```kie-drools-wb``` (both cloning a ```github``` Repository with ```SCP``` URL and cloning a Workbench repository with ```ssh://user@localhost:8001/uf-playground```). 

Steps to test:-

1) Create a new Repository at github.com
2) Create an SSH key; see [here](https://help.github.com/articles/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent/) (a)
3) Add your SSH key to github; see [here](https://help.github.com/articles/adding-a-new-ssh-key-to-your-github-account/)
4) Run ```kie-drools-wb``` (or which ever you fancy)
5) Clone a Repository using the ```git@github.com:<user>/<repositorty>.git``` URL

(a) *IF* you entered a passphrase when generating your public key you'll need to specify the passphrase with System Property ```org.uberfire.nio.git.ssh.passphrase```
